core/src/main/scala/filodb.core/memstore/TimeSeriesMemStore.scala

```diff
@@ -140,16 +140,22 @@ extends MemStore with StrictLogging {
   def recoverStream(dataset: DatasetRef,
                     shardNum: Int,
                     stream: Observable[SomeData],
+                    startOffset: Long,
+                    endOffset: Long,
                     checkpoints: Map[Int, Long],
                     reportingInterval: Long): Observable[Long] = {
     val shard = getShardE(dataset, shardNum)
-    val endOffset = checkpoints.values.max
     shard.setGroupWatermarks(checkpoints)
-    var targetOffset = checkpoints.values.min + reportingInterval
-    stream.map(shard.ingest(_)).collect {
-      case offset: Long if offset > targetOffset || offset >= endOffset =>
-        targetOffset += reportingInterval
-        offset
+    if (endOffset < startOffset) Observable.empty
```

Comment:
You don't need to pass in endOffset and startOffset, they are calculatable from checkpoints.

Reply:
I know - I wanted one place where the calculation was done. Hence the parameter is being passed in.